provider/balance_checker.go

```diff
@@ -83,6 +102,82 @@ func (bc *balanceChecker) doCheck(ctx context.Context) (bool, error) {
 	return tooLow, nil
 }
 
+func (bc *balanceChecker) runEscrowCheck(ctx context.Context, lid mtypes.LeaseID, res chan<- leaseCheckResponse) {
+	go func() {
+		select {
+		case <-bc.lc.Done():
+		case res <- bc.doEscrowCheck(ctx, lid):
+		}
+	}()
+}
+
+func (bc *balanceChecker) doEscrowCheck(ctx context.Context, lid mtypes.LeaseID) leaseCheckResponse {
+	resp := leaseCheckResponse{
+		lid: lid,
+	}
+
+	var syncInfo *tmrpc.SyncInfo
+	syncInfo, resp.err = bc.session.Client().NodeSyncInfo(ctx)
+	if resp.err != nil {
+		return resp
+	}
+
+	if !syncInfo.CatchingUp {
+		resp.err = aclient.ErrNodeNotSynced
+		return resp
+	}
+
+	var dResp *dtypes.QueryDeploymentResponse
+	var lResp *mtypes.QueryLeasesResponse
+
+	// Fetch the balance of the escrow account
+	dResp, resp.err = bc.aqc.Deployment(ctx, &dtypes.QueryDeploymentRequest{
+		ID: lid.DeploymentID(),
+	})
+
+	if resp.err != nil {
+		return resp
+	}
+
+	lResp, resp.err = bc.aqc.Leases(ctx, &mtypes.QueryLeasesRequest{
+		Filters: mtypes.LeaseFilters{
+			Owner: lid.Owner,
+			DSeq:  lid.DSeq,
+			State: "active",
+		},
+	})
+
+	if resp.err != nil {
+		return resp
+	}
+
+	totalLeaseAmount := sdk.NewDec(0)
+	for _, lease := range lResp.Leases {
+		totalLeaseAmount = totalLeaseAmount.Add(lease.Lease.Price.Amount)
+	}
+
+	if resp.err == nil {
+		balance := dResp.EscrowAccount.Balance.Amount
+		settledAt := dResp.EscrowAccount.SettledAt
+		balanceRemaining := balance.MustFloat64() - (float64(syncInfo.LatestBlockHeight-settledAt))*totalLeaseAmount.MustFloat64()
+		blocksRemaining := int64(balanceRemaining / totalLeaseAmount.MustFloat64())
+
+		// lease is out of funds
+		if blocksRemaining <= 0 {
+			return resp
+		}
+
+		blocksPerCheckInterval := int64(bc.cfg.LeaseFundCheckInterval / blockPeriod)
+		if blocksRemaining > blocksPerCheckInterval {
+			blocksRemaining = blocksPerCheckInterval
+		}
+
+		resp.checkAfter = time.Duration(blocksRemaining) * (6500 * time.Millisecond)
```

Comment:
We should add a constant for estimated block time somewhere, if we don't have that already

Reply:
yeah, it is right above

Reply:
Isn't `6500 * time.Millisecond` in this line some sort of rough estimate of block time?

Reply:
indeed, ideally, we want to have real block time

Reply:
there's a `blockPeriod` constant defined above.

I agree @hydrogen18 but I'm also ok with leaving it as a constant right now.

Reply:
Really, why don't we just use 5 seconds for the math here? It would just result in the check running consistently early in almost all cases. Which is fine.